data/slash/multipleHeaderSigningProof_test.go

```diff
@@ -33,78 +38,109 @@ func TestNewMultipleSigningProof(t *testing.T) {
 			expectedErr: data.ErrNilHeaderInfo,
 		},
 		{
-			args:        make(map[string]slash.SlashingResult),
-			expectedErr: nil,
+			args: map[string]slash.SlashingResult{
+				"pubKey": {Headers: []data.HeaderInfoHandler{
+					&dataMock.HeaderInfoStub{Header: &block.HeaderV2{}, Hash: []byte("h")},
+					&dataMock.HeaderInfoStub{Header: &block.HeaderV2{}, Hash: []byte("h")}},
+				},
+			},
+			expectedErr: data.ErrHeadersSameHash,
 		},
 	}
 
 	for _, currTest := range tests {
 		_, err := slash.NewMultipleSigningProof(currTest.args)
-		require.Equal(t, currTest.expectedErr, err)
+		require.Error(t, err)
+		require.True(t, strings.Contains(err.Error(), currTest.expectedErr.Error()))
 	}
 }
 
-func TestMultipleSigningProof_GetHeadersGetLevelGetType(t *testing.T) {
+func TestMultipleSigningProof_GetHeadersGetLevel(t *testing.T) {
 	h1 := &block.HeaderV2{Header: &block.Header{TimeStamp: 1}}
 	h2 := &block.HeaderV2{Header: &block.Header{TimeStamp: 2}}
 	h3 := &block.HeaderV2{Header: &block.Header{TimeStamp: 3}}
+	h4 := &block.HeaderV2{Header: &block.Header{TimeStamp: 4}}
+	h5 := &block.HeaderV2{Header: &block.Header{TimeStamp: 5}}
 
 	hInfo1 := &dataMock.HeaderInfoStub{Header: h1, Hash: []byte("h1")}
 	hInfo2 := &dataMock.HeaderInfoStub{Header: h2, Hash: []byte("h2")}
 	hInfo3 := &dataMock.HeaderInfoStub{Header: h3, Hash: []byte("h3")}
+	hInfo4 := &dataMock.HeaderInfoStub{Header: h4, Hash: []byte("h4")}
+	hInfo5 := &dataMock.HeaderInfoStub{Header: h5, Hash: []byte("h5")}
 
 	slashRes1 := slash.SlashingResult{
 		SlashingLevel: slash.High,
-		Headers:       []data.HeaderInfoHandler{hInfo1, hInfo2},
+		Headers:       []data.HeaderInfoHandler{hInfo4, hInfo2, hInfo1, hInfo3},
 	}
 	slashRes2 := slash.SlashingResult{
 		SlashingLevel: slash.Medium,
-		Headers:       []data.HeaderInfoHandler{hInfo3},
+		Headers:       []data.HeaderInfoHandler{hInfo3, hInfo5, hInfo4},
+	}
+	slashRes3 := slash.SlashingResult{
+		SlashingLevel: slash.Zero,
+		Headers:       []data.HeaderInfoHandler{},
+	}
+	slashRes4 := slash.SlashingResult{
+		SlashingLevel: slash.Zero,
+		Headers:       nil,
 	}
 	slashRes := map[string]slash.SlashingResult{
 		"pubKey1": slashRes1,
 		"pubKey2": slashRes2,
+		"pubKey3": slashRes3,
+		"pubKey4": slashRes4,
 	}
 
 	proof, err := slash.NewMultipleSigningProof(slashRes)
 	require.Nil(t, err)
 
-	require.Equal(t, slash.MultipleSigning, proof.GetType())
 	require.Equal(t, slash.High, proof.GetLevel([]byte("pubKey1")))
 	require.Equal(t, slash.Medium, proof.GetLevel([]byte("pubKey2")))
 	require.Equal(t, slash.Zero, proof.GetLevel([]byte("pubKey3")))
+	require.Equal(t, slash.Zero, proof.GetLevel([]byte("pubKey4")))
+	require.Equal(t, slash.Zero, proof.GetLevel([]byte("pubKey5")))
 
-	require.Len(t, proof.GetHeaders([]byte("pubKey1")), 2)
-	require.Len(t, proof.GetHeaders([]byte("pubKey2")), 1)
+	require.Len(t, proof.GetHeaders([]byte("pubKey1")), 4)
```

Comment:
can you create expected headers lists for the pubKeys and then check only with require.Equal?

Reply:
Yes, changed it accordingly ! :+1: